typed-racket-test/succeed/opaque-object-stronger.rkt

```diff
@@ -184,3 +185,21 @@
       (m1 (->m any/c any/c any/c))))
 
 )
+
+(let () ;; restrict-typed->/c and restrict-typed-field/c
```

Comment:
Can we add some tests with actual object contracts?

Reply:
I don't know how to write a test like that. You only get one of the restrictions when an opaque object contract is dynamically applied to an object with methods or fields not mentioned in the contract.

Reply:
How about running the example program with a big time & memory limit?

Reply:
Can you then write a contract showing that something gets wrapped only once (perhaps using `eq?`)?

Reply:
I'm not sure what a test case like that would look like, doing something like this:
```
(let ()
  (define ctc
    (object/c-opaque
     (mtd (->m any/c any/c))))
  (define o
    (new
     (class object%
       (super-new)
       (define/public (mtd x) x)
       (define/public (guts) #f))))
  (define oc (contract ctc o 'p 'n))
  (define oc2 (contract ctc oc 'p 'n))
  (define oc3 (contract ctc oc2 'p 'n))
  (define oc4 (contract ctc oc3 'p 'n))
  (eq? oc3 oc4))
```
doesn't work.

@rfindler do you have any ideas on writing a better test case for this?

Reply:
nope!

Reply:
What about using `contract-stronger?` on the results of `value-contract`?

Also, why does that test fail? Also, why do you need 4 uses of `contract` there -- shouldn't two be enough?

Reply:
I pushed a better test case using `contract-stronger?` and `value-contract`. I don't fully understand object contracts well enough to understand how the contract elimination strategy affects things overall.

Reply:
Awesome. @rfindler, do you know why @dfeltey's initial test didn't work?

Reply:
I think a reasonable test case calls `contract-stronger?` on the actual contract that gets generated (since that's what this fix is about). As for the test case above, I don't think it should be `eq?`?